pkg/cloudprovider/providers/azure/azure_storage_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+)
+
+func TestCreateFileShare(t *testing.T) {
+	cloud := &Cloud{}
+	fake := newFakeStorageAccountClient()
+	cloud.StorageAccountClient = fake
+	cloud.FileClient = &fakeFileClient{}
+
+	name := "baz"
+	sku := "sku"
+	location := "centralus"
+	value := "foo key"
+	bogus := "bogus"
+
+	tests := []struct {
+		name     string
+		acct     string
+		acctType string
+		loc      string
+		gb       int
+		accounts storage.AccountListResult
+		keys     storage.AccountListKeysResult
+		err      error
+
+		expectErr  bool
```

Comment:
It's better to test for a specific error message rather then just a fact of error. Otherwise, when new check will be added to the method, all the existing test cases could start failing on another error and we won't catch that.

Reply:
but testing on an error message is equally brittle, if someone changes the error text (makes it better or somesuch) all of the tests start to fail.

I feel like we are saying "we expect this call to fail" we're not saying "we expect this call to fail with this specific error" and it's ok to just test "we expect this call to fail"

As above, I'm happy enough to add it if you really insist, but I personally don't see the point.

Reply:
> but testing on an error message is equally brittle, if someone changes the error text (makes it better or somesuch) all of the tests start to fail.

I agree that it's brittle, but I can't agree that they're equally brittle :) Testing only for error could easily lead us to the situation when most of our tests don't actually test all the cases but only first one. In this case, we'll be sure that we're safe to do refactoring but we aren't. So my point is to avoid having unreliable tests.

Reply:
done.

Reply:
Thanks!
Hm.. I don't see them. Maybe you forgot to push some commits?

Reply:
I believe this part here:

https://github.com/kubernetes/kubernetes/pull/59165/files#diff-599cc8981f792662858b3656805661d4R155

Addresses your comment, let me know if there is more you want to see.

Reply:
I see that `TestProcessRoutes` has a code for checking the error message but it doesn't use it as all test cases test for success. At the same time, `TestCreateFileShare` could benefit from checking for error message but lack such a check. Looks like you misplaced the check to a wrong test?